pkg/client/testclient/fixture.go

```diff
@@ -216,3 +218,46 @@ func (o objects) Add(obj runtime.Object) error {
 
 	return nil
 }
+
+func (o objects) Delete(obj runtime.Object) error {
```

Comment:
This breaks clients that want to return errors from delete.  You probably need to take that into account (test cases depend on that behavior in other places).

Instead of implementing delete, you should be able to use staged gets.  What pattern of calls are you testing?  If so, you should be doing:

    # retrieve a pod
    o.Add(&Pod{})
    # get error
    o.Add(&Status{.... error})

If your testing code calls Delete then Get, they should get the right behavior.